wagtail/wagtailimages/views/serve.py

```diff
@@ -0,0 +1,68 @@
+from wsgiref.util import FileWrapper
+import imghdr
+import base64
+import hmac
+import hashlib
+
+from django.shortcuts import get_object_or_404
+from django.http import HttpResponse, StreamingHttpResponse, HttpResponsePermanentRedirect
+from django.core.exceptions import PermissionDenied, ImproperlyConfigured
+from django.views.generic import View
+from django.conf import settings
+from django.utils.six import text_type
+
+from wagtail.wagtailimages.models import get_image_model, SourceImageIOError
+from wagtail.wagtailimages.exceptions import InvalidFilterSpecError
+
+
+def generate_signature(image_id, filter_spec, key=None):
+    if key is None:
+        key = settings.SECRET_KEY
+
+    # Key must be a bytes object
+    if isinstance(key, text_type):
+        key = key.encode()
+
+    # Based on libthumbor hmac generation
+    # https://github.com/thumbor/libthumbor/blob/b19dc58cf84787e08c8e397ab322e86268bb4345/libthumbor/crypto.py#L50
+    url = str(image_id) + '/' + str(filter_spec) + '/'
```

Comment:
You wouldnt even need to call `str()` on the arguments, as `.format()` does that for you:

```python
url = '{}/{}/'.format(image_id, filter_spec)
```